script.js

```diff
@@ -1,43 +1,131 @@
-window.onload = function onload() { };
-
-function createProductImageElement(imageSource) {
-  const img = document.createElement('img');
-  img.className = 'item__image';
-  img.src = imageSource;
-  return img;
-}
-
-function createCustomElement(element, className, innerText) {
-  const e = document.createElement(element);
-  e.className = className;
-  e.innerText = innerText;
-  return e;
-}
-
-function createProductItemElement({ sku, name, image }) {
-  const section = document.createElement('section');
-  section.className = 'item';
-
-  section.appendChild(createCustomElement('span', 'item__sku', sku));
-  section.appendChild(createCustomElement('span', 'item__title', name));
-  section.appendChild(createProductImageElement(image));
-  section.appendChild(createCustomElement('button', 'item__add', 'Adicionar ao carrinho!'));
-
-  return section;
-}
-
-function getSkuFromProductItem(item) {
-  return item.querySelector('span.item__sku').innerText;
-}
-
-function cartItemClickListener(event) {
-  // coloque seu código aqui
-}
-
-function createCartItemElement({ sku, name, salePrice }) {
-  const li = document.createElement('li');
-  li.className = 'cart__item';
-  li.innerText = `SKU: ${sku} | NAME: ${name} | PRICE: $${salePrice}`;
-  li.addEventListener('click', cartItemClickListener);
-  return li;
-}
+  const getItemsStorage = () => {
+    let cartItems = [];
+    if (localStorage.getItem('cart')) {
+      cartItems = JSON.parse(localStorage.getItem('cart'));
+    }
+    return cartItems;
+  };
+
+  async function updateTotalPrice() {
+    try {
+      let totalPrice = 0;
+      const cartItems = await getItemsStorage();
+      totalPrice = cartItems.reduce((total, item) => total + item.price, 0);
+      document.querySelector('.total-price').innerHTML = `${totalPrice}`;
+    } catch (err) {
+      console.log(err.msg);
+    }
+  }
+
+  document.querySelector('.empty-cart').addEventListener('click', () => {
+    localStorage.setItem('cart', '');
+    document.querySelectorAll('ol li').forEach(li => li.remove());
+    updateTotalPrice();
+  });
+
+  // function getSkuFromProductItem(item) {
+  //   return item.querySelector('span.item__sku').innerText;
+  // }
+
+  function cartItemClickListener(event) {
+    // coloque seu código aqui
+    const item = event.target.innerText.split(' | ');
+    const id = item[0].substr(5);
+    const cartItems = getItemsStorage();
+    const excl = cartItems.find(i => i.id === id);
+    cartItems.splice(cartItems.indexOf(excl), 1);
+    localStorage.setItem('cart', JSON.stringify(cartItems));
```

Comment:
Sintaxe mais moderna para o localStorage
```suggestion
    localStorage.cart = JSON.stringify(cartItems);
```

Reply:
Boa Pantalena, fica mais limpo o codigo